agent/agents/supervisor/supervisor.go

```diff
@@ -473,6 +488,11 @@ func (s *Supervisor) startProcess(agentID string, agentProcess *agentpb.SetState
 		close(done)
 	}()
 
+	if !<-process.IsInitialized() {
+		defer cancel()
+		return process.GetError()
+	}
```

Comment:
The supervisor here is supposed to wait until the process update the initialization status. But I am ok with make it a non-blocking operation with a sleep so that the supervisor will not be blocked forever.

Reply:
we can revert last changes made here, I just realized that `initialized` will be returned anyway.

Reply:
I am ok to leave it there. Non-blocking operation for supervisord is a better design.